resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -77,8 +77,12 @@ private[spark] class Client(
 
   def run(): Unit = {
     logInfo(s"Starting application $kubernetesAppId in Kubernetes...")
-    val submitterLocalFiles = KubernetesFileUtils.getOnlySubmitterLocalFiles(sparkFiles)
-    val submitterLocalJars = KubernetesFileUtils.getOnlySubmitterLocalFiles(sparkJars)
+    val submitterLocalFiles = KubernetesFileUtils.getOnlySubmitterLocalFiles(sparkFiles
+      .map(_.split(","))
+      .getOrElse(Array.empty[String]))
+    val submitterLocalJars = KubernetesFileUtils.getOnlySubmitterLocalFiles(sparkJars
```

Comment:
the `sparkJars` and `sparkFiles` instance variables here are still in the comma-separated format which is a little unexpected.  They also don't have type annotations which maybe could've helped in review.

Maybe we should do the comma-splitting in the first access of `spark.files` and `spark.jars` and then annotate the types as `Iterable[String]` instead of the current implicit `Option[String]` ?

```
   private val sparkFiles = sparkConf.getOption("spark.files")
   private val sparkJars = sparkConf.getOption("spark.jars")
```